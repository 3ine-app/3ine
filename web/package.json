{
  "name": "web",
  "version": "0.0.0",
  "license": "MIT",
  "homepage": "https://3ine.netlify.com",
  "displayName": "3ine.app",
  "description": "3ine.app",
  "browserslist": {
    "development": [
      "last 1 version"
    ],
    "production": [
      "defaults",
      "not IE 11",
      "not IE_Mob 11"
    ]
  },
  "dependencies": {
    "@redwoodjs/forms": "^0.38.0",
    "@redwoodjs/router": "^0.38.0",
    "@redwoodjs/web": "^0.38.0",
    "daisyui": "^1.14.2",
    "medium-editor": "^5.23.3",
    "prop-types": "15.7.2",
    "react": "17.0.2",
    "react-dom": "17.0.2",
    "react-icons": "^4.2.0"
  },
  "devDependencies": {
    "@storybook/addon-actions": "^6.3.7",
    "@storybook/addon-controls": "^6.3.7",
    "@storybook/addon-docs": "^6.3.8",
    "@tailwindcss/typography": "^0.4.1",
    "@types/medium-editor": "^5.0.4",
    "autoprefixer": "^10.3.1",
    "postcss": "^8.3.6",
    "postcss-loader": "^6.1.1",
    "tailwindcss": "^2.2.7"
  },
  "keywords": [
    "wine",
    "wset",
    "react",
    "jamstack",
    "redwoodjs",
    "fullstack",
    "javascript",
    "typescript",
    "tailwindcss",
    "property-beacon"
  ]
}