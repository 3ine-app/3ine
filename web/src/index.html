<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta name="mobile-web-app-capable" content="yes" />
    <meta name="apple-mobile-web-app-capable" content="yes" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <link rel="manifest" href="/manifest.webmanifest" />
    <link rel="icon" type="image/png" href="/favicon.ico" />
  </head>

  <body>
    <div id="redwood-app">
      <!-- Please keep the line below for prerender support. -->
      <%= prerenderPlaceholder %>
    </div>
  </body>
</html>
