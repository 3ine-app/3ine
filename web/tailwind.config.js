module.exports = {
  mode: 'jit',
  purge: ['./src/**/*.{js,jsx,ts,tsx}'],
  darkMode: false, // or 'media' or 'class'
  theme: {
    extend: {
      fontFamily: {
        'open-sans': ['Open Sans', 'sans-serif'],
        gruppo: ['Gruppo', 'cursive'],
      },
    },
  },
  variants: {
    extend: {},
  },
  plugins: [require('@tailwindcss/typography'), require('daisyui')],
  daisyui: {
    themes: ['valentine'],
  },
}
