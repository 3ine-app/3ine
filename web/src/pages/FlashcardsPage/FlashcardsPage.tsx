import { MetaTags } from '@redwoodjs/web'
import { useState } from 'react'
import FlashCardsCell from 'src/components/FlashCardsCell'
import { mergeClassName } from 'src/utils'

// TODO: Move to global level
const TAGS = [
  'grape',
  'region',
  'colour',
  'SAT',
  'aroma',
  'terroir',
  'tasting',
  'labelling',
  'appellation',
  'stainless steel',
  'inert vessels',
  'fermentation',
  'food pairing',
  'temperature',
  'winemaking',
  'white',
  'red',
  'sweet',
  'chardonnay',
  'riesling',
  'sauvignon blanc',
  'palates',
  'blending',
  'vintage',
  'maturation',
  'secondary',
  'botrytis',
  'wine 101',
  'storage',
  'ageing',
  'sparkling',
  'fortified',
]
const FlashcardsPage = () => {
  const randomIndex = Math.floor(Math.random() * TAGS.length)
  const [currentTags, setCurrentTags] = useState(
    TAGS.slice(randomIndex, randomIndex + 1)
  )

  return (
    <>
      <MetaTags
        title="Flashcards"
        // description="Flashcards description"
        /* you should un-comment description and add a unique description, 155 characters or less
        You can look at this documentation for best practices : https://developers.google.com/search/docs/advanced/appearance/good-titles-snippets */
      />
      <div className="flex flex-wrap">
        <div className="collapse rounded-box collapse-arrow shadow-xl">
          <input type="checkbox" />
          <div className="collapse-title text-xl font-medium pr-8">
            {currentTags.map((tag, index) => (
              <div
                key={`badge-${index}`}
                className="badge shadow-md badge-sm md:badge-md lg:badge-lg capitalize mx-0.5"
              >
                {tag}
              </div>
            ))}
          </div>
          <div className="collapse-content flex flex-wrap">
            {TAGS.map((tag, index) => (
              <div className="form-control" key={`tag-checkbox-${index}`}>
                <label className="cursor-pointer label">
                  <input
                    type="checkbox"
                    name="flashCardTags"
                    checked={currentTags.includes(tag)}
                    onChange={() => {
                      if (currentTags.includes(tag)) {
                        setCurrentTags(currentTags.filter((t) => t !== tag))
                      } else {
                        setCurrentTags([...currentTags, tag])
                      }
                    }}
                    className="checkbox-xs md:checkbox-sm lg:checkbox-md checkbox checkbox-neutral rounded-full"
                  />
                  <span
                    className={mergeClassName(
                      'label-text capitalize ml-1 mr-2 text-xs md:text-sm',
                      currentTags.includes(tag) ? 'text-shadow' : ''
                    )}
                  >
                    {tag}
                  </span>
                </label>
              </div>
            ))}
          </div>
        </div>
      </div>
      <div className="mt-10 mb-20 flex justify-center">
        <FlashCardsCell tags={currentTags.map((t) => t.toLowerCase())} />
      </div>
    </>
  )
}

export default FlashcardsPage
