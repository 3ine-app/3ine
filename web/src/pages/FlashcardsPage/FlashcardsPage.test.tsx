import { render } from '@redwoodjs/testing/web'
import FlashcardsPage from './FlashcardsPage'

describe('FlashcardsPage', () => {
  it('renders successfully', () => {
    expect(() => {
      expect(render(<FlashcardsPage />)).toMatchSnapshot()
    }).not.toThrow()
  })
})
