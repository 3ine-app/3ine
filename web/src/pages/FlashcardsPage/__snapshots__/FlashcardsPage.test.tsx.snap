// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlashcardsPage renders successfully 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="carousel rounded-box shadow-2xl"
      >
        <div
          class="carousel-item"
        >
          <div
            class="stack my-8 mx-2 font-gruppo"
          >
            <button
              class="rounded-xl text-primary-content hover:bg-primary-focus bg-primary-focus"
            >
              <div
                class="py-6 text-shadow capitalize"
              >
                favorite
              </div>
            </button>
            <div
              class="text-center w-36 card bg-primary text-primary-content animate-pulse"
            >
              <div
                class="py-6 capitalize"
              >
                favorite
              </div>
            </div>
            <div
              class="text-center w-36 card bg-primary text-primary-content animate-pulse"
            >
              <div
                class="py-6 capitalize"
              >
                favorite
              </div>
            </div>
          </div>
        </div>
        <div
          class="carousel-item"
        >
          <div
            class="stack my-8 mx-2 font-gruppo"
          >
            <button
              class="rounded-xl text-primary-content hover:bg-primary-focus bg-primary"
            >
              <div
                class="py-6 text-shadow capitalize"
              >
                random
              </div>
            </button>
            <div
              class="text-center w-36 card bg-primary text-primary-content"
            >
              <div
                class="py-6 capitalize"
              >
                random
              </div>
            </div>
            <div
              class="text-center w-36 card bg-primary text-primary-content"
            >
              <div
                class="py-6 capitalize"
              >
                random
              </div>
            </div>
          </div>
        </div>
        <div
          class="carousel-item"
        >
          <div
            class="stack my-8 mx-2 font-gruppo"
          >
            <button
              class="rounded-xl text-primary-content hover:bg-primary-focus bg-primary"
            >
              <div
                class="py-6 text-shadow capitalize"
              >
                tasting
              </div>
            </button>
            <div
              class="text-center w-36 card bg-primary text-primary-content"
            >
              <div
                class="py-6 capitalize"
              >
                tasting
              </div>
            </div>
            <div
              class="text-center w-36 card bg-primary text-primary-content"
            >
              <div
                class="py-6 capitalize"
              >
                tasting
              </div>
            </div>
          </div>
        </div>
        <div
          class="carousel-item"
        >
          <div
            class="stack my-8 mx-2 font-gruppo"
          >
            <button
              class="rounded-xl text-primary-content hover:bg-primary-focus bg-primary"
            >
              <div
                class="py-6 text-shadow capitalize"
              >
                pairing
              </div>
            </button>
            <div
              class="text-center w-36 card bg-primary text-primary-content"
            >
              <div
                class="py-6 capitalize"
              >
                pairing
              </div>
            </div>
            <div
              class="text-center w-36 card bg-primary text-primary-content"
            >
              <div
                class="py-6 capitalize"
              >
                pairing
              </div>
            </div>
          </div>
        </div>
        <div
          class="carousel-item"
        >
          <div
            class="stack my-8 mx-2 font-gruppo"
          >
            <button
              class="rounded-xl text-primary-content hover:bg-primary-focus bg-primary"
            >
              <div
                class="py-6 text-shadow capitalize"
              >
                winemaking
              </div>
            </button>
            <div
              class="text-center w-36 card bg-primary text-primary-content"
            >
              <div
                class="py-6 capitalize"
              >
                winemaking
              </div>
            </div>
            <div
              class="text-center w-36 card bg-primary text-primary-content"
            >
              <div
                class="py-6 capitalize"
              >
                winemaking
              </div>
            </div>
          </div>
        </div>
        <div
          class="carousel-item"
        >
          <div
            class="stack my-8 mx-2 font-gruppo"
          >
            <button
              class="rounded-xl text-primary-content hover:bg-primary-focus bg-primary"
            >
              <div
                class="py-6 text-shadow capitalize"
              >
                white
              </div>
            </button>
            <div
              class="text-center w-36 card bg-primary text-primary-content"
            >
              <div
                class="py-6 capitalize"
              >
                white
              </div>
            </div>
            <div
              class="text-center w-36 card bg-primary text-primary-content"
            >
              <div
                class="py-6 capitalize"
              >
                white
              </div>
            </div>
          </div>
        </div>
        <div
          class="carousel-item"
        >
          <div
            class="stack my-8 mx-2 font-gruppo"
          >
            <button
              class="rounded-xl text-primary-content hover:bg-primary-focus bg-primary"
            >
              <div
                class="py-6 text-shadow capitalize"
              >
                red
              </div>
            </button>
            <div
              class="text-center w-36 card bg-primary text-primary-content"
            >
              <div
                class="py-6 capitalize"
              >
                red
              </div>
            </div>
            <div
              class="text-center w-36 card bg-primary text-primary-content"
            >
              <div
                class="py-6 capitalize"
              >
                red
              </div>
            </div>
          </div>
        </div>
        <div
          class="carousel-item"
        >
          <div
            class="stack my-8 mx-2 font-gruppo"
          >
            <button
              class="rounded-xl text-primary-content hover:bg-primary-focus bg-primary"
            >
              <div
                class="py-6 text-shadow capitalize"
              >
                sparkling
              </div>
            </button>
            <div
              class="text-center w-36 card bg-primary text-primary-content"
            >
              <div
                class="py-6 capitalize"
              >
                sparkling
              </div>
            </div>
            <div
              class="text-center w-36 card bg-primary text-primary-content"
            >
              <div
                class="py-6 capitalize"
              >
                sparkling
              </div>
            </div>
          </div>
        </div>
        <div
          class="carousel-item"
        >
          <div
            class="stack my-8 mx-2 font-gruppo"
          >
            <button
              class="rounded-xl text-primary-content hover:bg-primary-focus bg-primary"
            >
              <div
                class="py-6 text-shadow capitalize"
              >
                fortified
              </div>
            </button>
            <div
              class="text-center w-36 card bg-primary text-primary-content"
            >
              <div
                class="py-6 capitalize"
              >
                fortified
              </div>
            </div>
            <div
              class="text-center w-36 card bg-primary text-primary-content"
            >
              <div
                class="py-6 capitalize"
              >
                fortified
              </div>
            </div>
          </div>
        </div>
        <div
          class="carousel-item"
        >
          <div
            class="stack my-8 mx-2 font-gruppo"
          >
            <button
              class="rounded-xl text-primary-content hover:bg-primary-focus bg-primary"
            >
              <div
                class="py-6 text-shadow capitalize"
              >
                region
              </div>
            </button>
            <div
              class="text-center w-36 card bg-primary text-primary-content"
            >
              <div
                class="py-6 capitalize"
              >
                region
              </div>
            </div>
            <div
              class="text-center w-36 card bg-primary text-primary-content"
            >
              <div
                class="py-6 capitalize"
              >
                region
              </div>
            </div>
          </div>
        </div>
        <div
          class="carousel-item"
        >
          <div
            class="stack my-8 mx-2 font-gruppo"
          >
            <button
              class="rounded-xl text-primary-content hover:bg-primary-focus bg-primary"
            >
              <div
                class="py-6 text-shadow capitalize"
              >
                country
              </div>
            </button>
            <div
              class="text-center w-36 card bg-primary text-primary-content"
            >
              <div
                class="py-6 capitalize"
              >
                country
              </div>
            </div>
            <div
              class="text-center w-36 card bg-primary text-primary-content"
            >
              <div
                class="py-6 capitalize"
              >
                country
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="sm:flex sm:justify-center my-20"
      >
        <div
          class="carousel flashcards rounded-box bg-gradient-to-t from-neutral to-primary"
        >
          <div
            class="flashcard bg-base-100 mx-auto mt-5 rounded-xl animate-pulse"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="carousel rounded-box shadow-2xl"
    >
      <div
        class="carousel-item"
      >
        <div
          class="stack my-8 mx-2 font-gruppo"
        >
          <button
            class="rounded-xl text-primary-content hover:bg-primary-focus bg-primary-focus"
          >
            <div
              class="py-6 text-shadow capitalize"
            >
              favorite
            </div>
          </button>
          <div
            class="text-center w-36 card bg-primary text-primary-content animate-pulse"
          >
            <div
              class="py-6 capitalize"
            >
              favorite
            </div>
          </div>
          <div
            class="text-center w-36 card bg-primary text-primary-content animate-pulse"
          >
            <div
              class="py-6 capitalize"
            >
              favorite
            </div>
          </div>
        </div>
      </div>
      <div
        class="carousel-item"
      >
        <div
          class="stack my-8 mx-2 font-gruppo"
        >
          <button
            class="rounded-xl text-primary-content hover:bg-primary-focus bg-primary"
          >
            <div
              class="py-6 text-shadow capitalize"
            >
              random
            </div>
          </button>
          <div
            class="text-center w-36 card bg-primary text-primary-content"
          >
            <div
              class="py-6 capitalize"
            >
              random
            </div>
          </div>
          <div
            class="text-center w-36 card bg-primary text-primary-content"
          >
            <div
              class="py-6 capitalize"
            >
              random
            </div>
          </div>
        </div>
      </div>
      <div
        class="carousel-item"
      >
        <div
          class="stack my-8 mx-2 font-gruppo"
        >
          <button
            class="rounded-xl text-primary-content hover:bg-primary-focus bg-primary"
          >
            <div
              class="py-6 text-shadow capitalize"
            >
              tasting
            </div>
          </button>
          <div
            class="text-center w-36 card bg-primary text-primary-content"
          >
            <div
              class="py-6 capitalize"
            >
              tasting
            </div>
          </div>
          <div
            class="text-center w-36 card bg-primary text-primary-content"
          >
            <div
              class="py-6 capitalize"
            >
              tasting
            </div>
          </div>
        </div>
      </div>
      <div
        class="carousel-item"
      >
        <div
          class="stack my-8 mx-2 font-gruppo"
        >
          <button
            class="rounded-xl text-primary-content hover:bg-primary-focus bg-primary"
          >
            <div
              class="py-6 text-shadow capitalize"
            >
              pairing
            </div>
          </button>
          <div
            class="text-center w-36 card bg-primary text-primary-content"
          >
            <div
              class="py-6 capitalize"
            >
              pairing
            </div>
          </div>
          <div
            class="text-center w-36 card bg-primary text-primary-content"
          >
            <div
              class="py-6 capitalize"
            >
              pairing
            </div>
          </div>
        </div>
      </div>
      <div
        class="carousel-item"
      >
        <div
          class="stack my-8 mx-2 font-gruppo"
        >
          <button
            class="rounded-xl text-primary-content hover:bg-primary-focus bg-primary"
          >
            <div
              class="py-6 text-shadow capitalize"
            >
              winemaking
            </div>
          </button>
          <div
            class="text-center w-36 card bg-primary text-primary-content"
          >
            <div
              class="py-6 capitalize"
            >
              winemaking
            </div>
          </div>
          <div
            class="text-center w-36 card bg-primary text-primary-content"
          >
            <div
              class="py-6 capitalize"
            >
              winemaking
            </div>
          </div>
        </div>
      </div>
      <div
        class="carousel-item"
      >
        <div
          class="stack my-8 mx-2 font-gruppo"
        >
          <button
            class="rounded-xl text-primary-content hover:bg-primary-focus bg-primary"
          >
            <div
              class="py-6 text-shadow capitalize"
            >
              white
            </div>
          </button>
          <div
            class="text-center w-36 card bg-primary text-primary-content"
          >
            <div
              class="py-6 capitalize"
            >
              white
            </div>
          </div>
          <div
            class="text-center w-36 card bg-primary text-primary-content"
          >
            <div
              class="py-6 capitalize"
            >
              white
            </div>
          </div>
        </div>
      </div>
      <div
        class="carousel-item"
      >
        <div
          class="stack my-8 mx-2 font-gruppo"
        >
          <button
            class="rounded-xl text-primary-content hover:bg-primary-focus bg-primary"
          >
            <div
              class="py-6 text-shadow capitalize"
            >
              red
            </div>
          </button>
          <div
            class="text-center w-36 card bg-primary text-primary-content"
          >
            <div
              class="py-6 capitalize"
            >
              red
            </div>
          </div>
          <div
            class="text-center w-36 card bg-primary text-primary-content"
          >
            <div
              class="py-6 capitalize"
            >
              red
            </div>
          </div>
        </div>
      </div>
      <div
        class="carousel-item"
      >
        <div
          class="stack my-8 mx-2 font-gruppo"
        >
          <button
            class="rounded-xl text-primary-content hover:bg-primary-focus bg-primary"
          >
            <div
              class="py-6 text-shadow capitalize"
            >
              sparkling
            </div>
          </button>
          <div
            class="text-center w-36 card bg-primary text-primary-content"
          >
            <div
              class="py-6 capitalize"
            >
              sparkling
            </div>
          </div>
          <div
            class="text-center w-36 card bg-primary text-primary-content"
          >
            <div
              class="py-6 capitalize"
            >
              sparkling
            </div>
          </div>
        </div>
      </div>
      <div
        class="carousel-item"
      >
        <div
          class="stack my-8 mx-2 font-gruppo"
        >
          <button
            class="rounded-xl text-primary-content hover:bg-primary-focus bg-primary"
          >
            <div
              class="py-6 text-shadow capitalize"
            >
              fortified
            </div>
          </button>
          <div
            class="text-center w-36 card bg-primary text-primary-content"
          >
            <div
              class="py-6 capitalize"
            >
              fortified
            </div>
          </div>
          <div
            class="text-center w-36 card bg-primary text-primary-content"
          >
            <div
              class="py-6 capitalize"
            >
              fortified
            </div>
          </div>
        </div>
      </div>
      <div
        class="carousel-item"
      >
        <div
          class="stack my-8 mx-2 font-gruppo"
        >
          <button
            class="rounded-xl text-primary-content hover:bg-primary-focus bg-primary"
          >
            <div
              class="py-6 text-shadow capitalize"
            >
              region
            </div>
          </button>
          <div
            class="text-center w-36 card bg-primary text-primary-content"
          >
            <div
              class="py-6 capitalize"
            >
              region
            </div>
          </div>
          <div
            class="text-center w-36 card bg-primary text-primary-content"
          >
            <div
              class="py-6 capitalize"
            >
              region
            </div>
          </div>
        </div>
      </div>
      <div
        class="carousel-item"
      >
        <div
          class="stack my-8 mx-2 font-gruppo"
        >
          <button
            class="rounded-xl text-primary-content hover:bg-primary-focus bg-primary"
          >
            <div
              class="py-6 text-shadow capitalize"
            >
              country
            </div>
          </button>
          <div
            class="text-center w-36 card bg-primary text-primary-content"
          >
            <div
              class="py-6 capitalize"
            >
              country
            </div>
          </div>
          <div
            class="text-center w-36 card bg-primary text-primary-content"
          >
            <div
              class="py-6 capitalize"
            >
              country
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="sm:flex sm:justify-center my-20"
    >
      <div
        class="carousel flashcards rounded-box bg-gradient-to-t from-neutral to-primary"
      >
        <div
          class="flashcard bg-base-100 mx-auto mt-5 rounded-xl animate-pulse"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
